### PR TITLE
Move state: Move state and handlers from Form component to GetDetails component. 

### PR DESCRIPTION
**Pull request by orjwan98**
What I did in this PR: References issue #129 #80 

**client/src/components/GetDetails/index.js**
- Moved handlers and all state from `Form` to `GetDetails` component (except for categories array) and passed them to `Form` as props to be used as they were before. 

- Set the initial values from Clarifai for the states in `selected_itemType` and `selected_colors`, and set the initial values for selected_category from data arrays using `componentDidMount`.

**client/src/components/Shared/Form/index.js**
- Imported handlers and states from `GetDetails` component to `Form` component through props and used them as they were before.
- Removed `data` prop from Popup component to be populated once the JSON file is ready.

### CODE REVIEW

File: client/src/components/GetDetails/index.js [hunk 1]
```diff
@@ -19,13 +30,49 @@ class GetDetails extends Component {
...
+  };
+
   render() {
     const { image_url, apparel, colors } = this.props.location.details;
     return (
       <React.Fragment>
         <Title {...this.props} />
         <Header title="Get your details" />
-        <Form image={image_url} apparel={apparel} colors={colors} />
+        <Form
```
Comment (jema28) [inline client/src/components/GetDetails/index.js:128]: Try passing down all state (see below) with {...this.state} and then destructuring them in Form.
```
          isOpen={this.state.isOpen}
          selected_itemType={this.state.selected_itemType}
          selected_colors={this.state.selected_colors}
          selected_brands={this.state.selected_brands}
          selected_condition={this.state.selected_condition}
          selected_labelSize={this.state.selected_labelSize}
          selected_age={this.state.selected_age}
          selectedCat={this.state.selectedCat}
          toggleOpen={this.toggleOpen}
          toggleClose={this.toggleClose}
          changeSelected={this.changeSelected}
```

File: client/src/components/Shared/Form/index.js [hunk 2]
```diff
@@ -73,42 +65,29 @@ class Form extends Component {
...
-  toggleOpen = (e) => {
-    const clicked = e.target.value.split('.');
-    if (clicked[1] === 'more') {
-      this.setState({ isOpen: true, selectedCat: clicked[0] });
-    } else {
-      const selected = `selected_${clicked[0]}`;
-      this.setState({ [selected]: e.target.value });
+      this.setState({
+        // selected_itemType: apparel[0],
+        // selected_colors: colors[0],
```
Comment (Hadeel-Salamin) [inline client/src/components/Shared/Form/index.js]: remove coding comments

File: client/src/components/Shared/Form/index.js [hunk 3]
```diff
@@ -73,42 +65,29 @@ class Form extends Component {
   };
 
   componentDidMount() {
-    const apparel = this.props.apparel.data.map(ele => ele.tag_name);
+    if (this.props.apparel) {
```
Comment (jema28) [inline client/src/components/Shared/Form/index.js]: You can destructure apparel and color from props here.

**Review by jema28 — CHANGES_REQUESTED** [review]

**Review by jema28 — APPROVED** [review]
Working well! 👍nice destructuring.

**Review by amanshawar — APPROVED** [review]

**Review by amanshawar — APPROVED** [review]

**Review by Hadeel-Salamin — APPROVED** [review]
Well done :+1:

**Review by FidaaHersh — APPROVED** [review]

**Review by jema28 — APPROVED** [review]

**Pull request merged**